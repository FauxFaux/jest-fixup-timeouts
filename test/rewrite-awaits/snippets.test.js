import { transform } from '@babel/core';

function runPlugin(code) {
  return transform(code, {
    babelrc: false,
    filename: 'test.js',
    plugins: [__dirname + '/../../lib/rewrite-awaits'],
  });
}

describe('translation of snippets', () => {
  it('transforms a bare function call in statement context', () => {
    // language=JavaScript
    const code = 'async function foo() { await bar(); }';
    expect(runPlugin(code).code).toMatchSnapshot();
  });

  it('transforms an expression', () => {
    // language=JavaScript
    const code = 'async function foo() { return await bar(); }';
    expect(runPlugin(code).code).toMatchSnapshot();
  });

  it('transforms an existing expect', () => {
    // language=JavaScript
    const code =
      'async function foo() {' +
      ' await expect(bar()).resolves.toBe("hot potatoes");' +
      '}';
    expect(runPlugin(code).code).toMatchSnapshot();
  });

  it('skips double stacking', () => {
    // language=JavaScript
    const code = 'async function foo() { await expect.withinDeadline(bar()); }';
    expect(runPlugin(code).code).toMatchSnapshot();
  });

  it('transforms multiple awaits', () => {
    // language=JavaScript
    const code =
      'async function foo() {' +
      ' await bar(1, 2, 3);' +
      ' await quux(1, 2, 3);' +
      '}';
    expect(runPlugin(code).code).toMatchSnapshot();
  });

  it('transforms await in argument context', () => {
    // language=JavaScript
    const code = 'async function foo() { await bar(1, await quux(), 3); }';
    expect(runPlugin(code).code).toMatchSnapshot();
  });
});
