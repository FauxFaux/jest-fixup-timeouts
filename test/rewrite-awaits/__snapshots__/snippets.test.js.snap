// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`translation of snippets skips double stacking 1`] = `
"\\"use strict\\";

async function foo() {
  await expect.withinDeadline(bar());
}"
`;

exports[`translation of snippets transforms a bare function call in statement context 1`] = `
"\\"use strict\\";

async function foo() {
  await expect.withinDeadline(bar());
}"
`;

exports[`translation of snippets transforms an existing expect 1`] = `
"\\"use strict\\";

async function foo() {
  await expect.withinDeadline(expect(bar()).resolves.toBe(\\"hot potatoes\\"));
}"
`;

exports[`translation of snippets transforms an expression 1`] = `
"\\"use strict\\";

async function foo() {
  return await expect.withinDeadline(bar());
}"
`;

exports[`translation of snippets transforms await in argument context 1`] = `
"\\"use strict\\";

async function foo() {
  await expect.withinDeadline(bar(1, await expect.withinDeadline(quux()), 3));
}"
`;

exports[`translation of snippets transforms multiple awaits 1`] = `
"\\"use strict\\";

async function foo() {
  await expect.withinDeadline(bar(1, 2, 3));
  await expect.withinDeadline(quux(1, 2, 3));
}"
`;
